src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -265,6 +266,7 @@ export class MeetingManager implements AudioVideoObserver {
 
       this.audioVideo.stop();
     }
+    this.eventController?.removeObserver(this.eventDidReceiveRef);
```

Comment:
@devalevenkatesh Should this be done in unmount event instead of leave ?

Reply:
I would check once whether we need to maintain the `eventDidReceiveRef` anymore. Earlier it was tied to `audioVideo` hence we ran into clean up issues. But we can check if the ref is needed or simply add when the `MeetingSession` is constructed and remove when it is destroyed. But still I think should be tied to `MeetingSession` lifecycle as we are getting the `eventController` from `join` method.

Reply:
`audioVideoDidStop()` calls `leave()` so it will cover those cases and I feel it's more intuitive to remove the observer in the meeting leave.

Reply:
Hmm, so the `leave` and the `audioVideoDidStop` logic is not intuitive.

On local attendee meeting leave:
The app will call meetingManager.leave() -> audioVideo.stop() -> JS SDK -> audioVideoDidStop() -> Get update in audioVideoDidStop() in react component library.

On a attendee ending a meeting (not leaving):
- JS SDK gets meeting ended from backend -> calls audioVideoDidStop on obsevers -> component library gets the update in audioVideoDidStop -> then calls meetingManager.leave() making the local attendee auto leave on meeting end.

so my concern is when a local attendee leaves and if we remove the observer right away before getting the final update in audioVideoDidStop we may loose some events.

Reply:
Ah that makes sense, yeah we can put it in `audioVideoDidStop()` then.